api/src/opentrons/protocols/advanced_control/transfers.py

```diff
@@ -633,6 +645,7 @@ def _expand_for_volume_constraints(
         """Split a sequence of proposed transfers if necessary to keep each
         transfer under the given max volume.
         """
+        assert max_volume > 0
```

Comment:
Let's leave a comment here to explain the intent:

```suggestion
        # A final defense against an infinite loop.
        # Raising a proper exception with a helpful message is left to calling code,
        # because it has more context about what the user is trying to do.
        assert max_volume > 0
```